Yearn-agents/PPSValues/src/agent.ts

```diff
@@ -0,0 +1,140 @@
+import BigNumber from "bignumber.js";
+import {
+  BlockEvent,
+  Finding,
+  HandleBlock,
+  FindingSeverity,
+  FindingType,
+  getJsonRpcUrl,
+} from "forta-agent";
+import Web3 from "web3";
+import { vaultAbi, getYearnVaults } from "./utils";
+import { CacheContainer } from "node-ts-cache";
+import { MemoryStorage } from "node-ts-cache-storage-memory";
+
+const web3 = new Web3(getJsonRpcUrl());
+
+export const createFinding = (
+  pps: string,
+  tracker: string,
+  reason: string,
+  id: number
+) => {
+  return Finding.fromObject({
+    name: "Yearn PPS Agent",
+    description: `Year PPS value: ${reason}`,
+    alertId: `Yearn-8-${id}`,
+    severity: FindingSeverity.High,
+    type: FindingType.Suspicious,
+    metadata: {
+      pps,
+      tracker,
+    },
+  });
+};
+
+interface CacheObject {
+  [key: string]: BigNumber;
+}
+
+const provideHandleFunction = (web3: Web3): HandleBlock => {
+  const threshold = 0.1;
+  const cache = new CacheContainer(new MemoryStorage());
+
+  return async (blockEvent: BlockEvent) => {
+    const findings: Finding[] = [];
+    let blockNumber = blockEvent.blockNumber;
+
+    const vaults = await getYearnVaults(web3, blockNumber);
```

Comment:
I think you should use cache mechanism here to fetch vaults per block. You can take a look at the Mauricio's PR for example.

Reply:
I can in id,e situation cache vaults too like the rest, but I was of impressions to check each block for new blocks too. Besides pops will change every block so I have to do the rest of the calls anyway,

Reply:
I see, then LGTM!